### PR TITLE
Wip/split temp basals

### PR DESCRIPTION
**Pull request by sulkaharo**
Split temp basals based on basal profile changes to get more accurate IOB
Tests for basal logic

NOTE: REQUIRES moment.js; run npm install to use if on linked install

DO NOT MERGE YET

### CODE REVIEW

**Comment by sulkaharo** [comment]
IOB from this branch vs "original" oref0 vs mmhistorytools

plot

**Comment by sulkaharo** [comment]
IMO this is now good to go. The IOB plot matches as expected (see above); tests pass. @jasoncalabrese or @scottleibrand you wanna do the honors?

File: tests/iob.test.js [hunk 1]
```diff
@@ -65,6 +67,7 @@ describe('IOB', function ( ) {
 
   it('should calculate IOB with Temp Basals', function() {
 
+	var basalprofile = [{'i': 0, 'start': '00:00:00', 'rate': 1, 'minutes': 0}];
```
Comment (scottleibrand) [inline tests/iob.test.js]: indentation looks a bit off.  tabs vs. spaces?

**Comment by scottleibrand** [comment]
I see some weird indentation that probably indicates a tabs vs. spaces mismatch, but otherwise LGTM.   @jasoncalabrese would you like to look over this and/or run it on a rig before we merge?

**Comment by sulkaharo** [comment]
FWIW we've had this on the rig for 5 days now; so far behaving as expected.

**Comment by scottleibrand** [comment]
Just pushed https://github.com/openaps/oref0/pull/202 to fix the inconsistent tab spacing.  I'm good to merge this via https://github.com/openaps/oref0/pull/202 as soon as someone else gives it a 👍 as well.

**Pull request merged**